frontend/system/src/Admin/Configuration/OptionEditModal.tsx

```diff
@@ -31,34 +30,44 @@ export const OptionEditModal = ({
   config,
   mode,
   category,
+  type,
   onClose,
 }: OptionEditModalProps) => {
   const t = useTranslation('system');
-  const uuid = useUuid();
 
-  const [value, setValue] = useState(config?.value ?? '');
+  const [value, setValue] = useState(config?.name ?? '');
+  const [addEntity, invalidateQueries] = useAddConfigItem();
 
-  const { Modal } = useDialog({ isOpen, onClose });
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
 
-  // TODO: set from queries
-  const isLoading = false;
+  const { Modal } = useDialog({ isOpen, onClose });
 
   return (
     <Modal
       okButton={
         <LoadingButton
           onClick={() => {
-            // TODO: save and handle errors
-            console.log('value to be saved:', {
-              id: config?.id ?? uuid(),
-              label: value,
-              value,
-            });
-            // if (!err || !err.message) {
-            //   err = { message: t('messages.unknown-error') };
-            // }
-            // setErrorMessage(err.message);
-            onClose();
+            setIsLoading(true);
+            addEntity({
+              input: {
+                name: value,
+                type: type,
+              },
+            })
+              .catch(err => {
```

Comment:
Should we disable this button for existing records and force a delete/insert?
Or should we allow editing/upsert?

Reply:
I could get behind add-only, though being able to fix typos is nice... if we do allow editing, might be good to rename the mutation to upsert rather than add?

Given changes to the entities are add-only as well I think that would be fine though!

Maybe rather than disabling the edit button once the modal is open, just remove the onclick on the table rows that opens the edit modal in the first place? At least until we have anything additional/interesting to display in the modal...